types/aws-lambda/index.d.ts

```diff
@@ -501,6 +501,52 @@ export interface AuthResponseContext {
     [name: string]: any;
 }
 
+/**
+ * CodePipeline events
+ * https://docs.aws.amazon.com/codepipeline/latest/userguide/actions-invoke-lambda-function.html
+ */
+export interface S3ArtifactLocation {
+    bucketName: string;
+    objectKey: string;
+}
+export interface S3ArtifactStore {
+    type: 'S3';
+    s3Location: S3ArtifactLocation;
+}
+
+export type ArtifactLocation = S3ArtifactStore;
+
+export interface Artifact {
+    name: string;
+    revision: string | null;
+    location: ArtifactLocation;
+}
```

Comment:
These names are a little generic maybe? I'd prefer `CodePipelineArtifact` etc, but I'm not sure what the `S3Artifact*` stuff should be named.
`Credentials` is fine, as it's a generic type (and I'm suprised we don't already have it)

Reply:
[Artifact](https://docs.aws.amazon.com/codepipeline/latest/APIReference/API_Artifact.html) docs mark all these properties as optional? I don't know how that works.

Reply:
I'll leave the names of the types as is -- they do actually line up with the API reference.  I've left them as mandatory, as I always saw them provided.